### DISCUSSION TITLE
Cyberduck not compatible with a self hosted Ceph S3 Storage?

### DISCUSSION THREAD

**Discussion by rc-networks** [iterate-ch/cyberduck#16548] [Q&A]
We are looking for a way to provide some kind of GUI for our self hosted Ceph S3 storage and someone suggested to use Cyberduck, so I'm here trying to figure out if it's gonna work.

So far, I can list the buckets:
<img width="637" alt="image" src="https://github.com/user-attachments/assets/52ef6716-f0de-4a95-a1e6-5caa6357d0ed">

but not their contents:
<img width="627" alt="image" src="https://github.com/user-attachments/assets/9048cedc-b5b4-45e9-915b-5c146ad62f87">

I've seen other articles/discussions about the same error:
```
Failed to parse XML document with handler class
```
but no working solution was found so far.

Can anyone shed some light please?

**Comment by AliveDevil** [comment]
I'm running Ceph S3, and didn't have any issue.
Two things here: Does your setup allow for Virtual Host/subdomain bucket access? i.e. bucket.radosgw.domain.internal, or is it only configured to allow subdirectory access (i.e. radosgw.domain.internal/bucket)?
In case only subdirectory access is configured: Use the `S3 (HTTPS, Deprecated Path Style)`-profile.

For reference, this is my S3 endpoint for Ceph which works with the `S3 (HTTPS)`
```
[client.rgw]
        rgw frontends = beast port=80
        rgw enable apis = s3
        rgw enable static website = true
        rgw trust forwarded https = true
        rgw resolve cname = true
```

<details>
<summary>radosgw-admin zonegroup get</summary>

```
{
    "id": "d6255542-f841-4121-adb6-66715a0bebcb",
    "name": "default",
    "api_name": "default",
    "is_master": true,
    "endpoints": [],
    "hostnames": [
        "s3.lab.xc",
        "swift.lab.xc",
        "s3.cloud.domain.internal",
        "swift.cloud.domain.internal",
        "s3-testing.lab.xc"
    ],
    "hostnames_s3website": [
        "s3-website.lab.xc",
        "s3-website.cloud.domain.internal"
    ],
    "master_zone": "55aa56aa-df95-4134-bdf3-6e70ab54c301",
    "zones": [
        {
            "id": "55aa56aa-df95-4134-bdf3-6e70ab54c301",
            "name": "default",
            "endpoints": [],
            "log_meta": false,
            "log_data": false,
            "bucket_index_max_shards": 11,
            "read_only": false,
            "tier_type": "",
            "sync_from_all": true,
            "sync_from": [],
            "redirect_zone": "",
            "supported_features": []
        }
    ],
    "placement_targets": [
        {
            "name": "default-placement",
            "tags": [],
            "storage_classes": [
                "GLACIER",
                "REDUCED_REDUNDANCY",
                "STANDARD"
            ]
        }
    ],
    "default_placement": "default-placement",
    "realm_id": "",
    "sync_policy": {
        "groups": []
    },
    "enabled_features": []
}
```

</details>

Required DNS records:
|Name|Type|Data|
|---|---|---|
|*.radosgw.domain.internal|CNAME|radosgw.domain.internal|
|radosgw.domain.internal|A|IP of RadosGW|

But this may create issues with multi-label bucket names (i.e. `some.website` would be `some.website.radosgw.domain.internal`, which a SSL wildcard certificate won't cover).
